src/reducers/uploadReducer.js

```diff
@@ -93,37 +98,162 @@ export function UploadReducer(state = { ...defaultUploadState }, action) {
 
       // Initiates files sent to upload area with status "uploading"
       // NOTE: Filtering done by ID, but in current implementation ID is set to FILENAME
-      const uploadingFiles = [...state.files].map(file => ({ file, status: 'UPLOADING', id: file.name }));
+      const uploadingFiles = [...state.stagedFiles].map(file => ({ file, status: 'UPLOADING', id: file.name }));
 
       // Creates dictionary from form fields
       const formData = {
         dataType: action.elements.dataType.value,
-        identifier: action.elements.identifier.value,
+        biospecimenID: action.elements.biospecimenID.value,
         collectionDate: action.elements.collectionDate.value,
         subjectType: action.elements.subjectType.value,
         studyPhase: action.elements.studyPhase.value,
         rawData: action.elements.rawData.checked,
         processedData: action.elements.processedData.checked,
       };
 
+      let expIndex = state.previousUploads
+        .findIndex(exp => ((exp.biospecimenID === formData.biospecimenID) && (exp.dataType === formData.dataType)));
+
+      const now = Date.now();
+      const d = new Date(now);
+
+      let prevUploads = [...state.previousUploads];
+
+      if (expIndex === -1) {
+        expIndex = 0;
+        const uploadDate = (d.getMonth()+1) + '/' + d.getDate() + '/' + d.getFullYear();
```

Comment:
Would it make sense for us to use a date/time manipulation npm package to create UNIX timestamps that are stored in the DB and to parse UNIX timestamps for UI rendering?

I had previously used `moment` (e.g. https://www.npmjs.com/package/moment) for similar purpose, but I don't prefer using it because it is sizable module to import. On the other hand, `dayjs` (e.g. https://www.npmjs.com/package/dayjs) is a much smaller module.